web/ui/react-app/src/Graph.tsx

```diff
@@ -238,64 +236,80 @@ class Graph extends PureComponent<GraphProps, GraphState> {
     if (!this.chartRef.current) {
       return;
     }
-    const selectedData = this.getData()[this.state.selectedSeriesIndex!];
     this.destroyPlot();
-    $.plot($(this.chartRef.current), selectedData ? [selectedData] : this.getData(), this.getOptions());
+
+    this.$chart = $.plot($(this.chartRef.current), this.state.chartData, this.getOptions());
   };
 
-  destroyPlot() {
-    const chart = $(this.chartRef.current!).data('plot');
-    if (chart !== undefined) {
-      chart.destroy();
-    }
+  destroyPlot = () => {
+    this.$chart.destroy && this.$chart.destroy();
```

Comment:
Shouldn't this be:

```suggestion
    this.$chart && this.$chart.destroy();
```

...as if there is a chart, it will always have a `destroy()` method, but there may not always be a chart?

Reply:
The approach is totally wrong. Initially, $chart should be undefined and not {}, thus checks here and there should be added but it will be more clear. I'll fix it.